plugin-server/src/cdp/templates/_transformations/property-filter/property-filter.template.ts

```diff
@@ -0,0 +1,166 @@
+            }
+        }
+        
+        if (shouldKeep) {
+            let value := obj[key]
+            // Recursively filter nested objects
+            if (typeof(value) = 'object' and value != null) {
+                result[key] := filterObject(value)
+            } else {
+                result[key] := value
+            }
+        }
+    }
+    
+    return result
+}
+
+// Helper function to filter properties recursively, but only in regular properties
+fun filterProperties(obj) {
+    if (obj = null) {
+        return null
+    }
+    
+    if (typeof(obj) != 'object') {
+        return obj
+    }
+    
+    let result := {}
+    let objKeys := keys(obj)
+    
+    for (let key in objKeys) {
+        let shouldKeep := true
+        
+        for (let propToFilter in propertiesToFilter) {
+            if (propertyNamesMatch(key, propToFilter)) {
+                shouldKeep := false
+            }
+        }
+        
+        if (shouldKeep) {
+            let value := obj[key]
+            if (key = '$set') {
+                if (includeSetProperties) {
+                    result[key] := filterObject(value)  // Use filterObject for $set
+                } else {
+                    result[key] := value
+                }
+            } else if (key = '$set_once') {
+                if (includeSetOnceProperties) {
+                    result[key] := filterObject(value)  // Use filterObject for $set_once
+                } else {
+                    result[key] := value
+                }
+            } else if (typeof(value) = 'object' and value != null) {
+                result[key] := filterProperties(value)
+            } else {
+                result[key] := value
+            }
+        }
+    }
+    
+    return result
+}
+
+// Create a copy of the event
+let returnEvent := {}
+
+// Copy non-filtered top-level properties
+let eventKeys := keys(event)
+for (let key in eventKeys) {
+    let shouldKeep := true
+    for (let propToFilter in propertiesToFilter) {
+        if (propertyNamesMatch(key, propToFilter)) {
+            shouldKeep := false
+        }
+    }
+    
+    if (shouldKeep) {
+        if (key = 'properties') {
+            returnEvent.properties := filterProperties(event.properties)
+        } else {
+            returnEvent[key] := event[key]
+        }
+    }
+}
+
+return returnEvent
+`,
+    inputs_schema: [
+        {
+            key: 'propertiesToFilter',
+            type: 'string',
+            label: 'Properties to Filter',
+            description: 'Comma-separated list of properties to filter (e.g. "ip,distinct_id,$ip")',
+            default: '$ip',
+            secret: false,
+            required: true,
+        },
+        {
+            key: 'includeSetProperties',
+            type: 'boolean',
+            label: 'Include $set properties',
+            description: 'If enabled, will also remove matching properties from $set object',
+            default: false,
```

Comment:
I think we should default these to true tbh as it is likely the most common situation